lib/main.dart

```diff
@@ -1,113 +1,279 @@
+import 'package:http/http.dart' as http;
+import 'package:flutter_userapp_demo/models/user.dart';
+import 'package:flutter_userapp_demo/main2.dart';
+import 'package:flutter_userapp_demo/custom_widgets/Pagination.dart';
+import 'package:flutter_userapp_demo/custom_widgets/user_card.dart';
+import 'package:flutter_userapp_demo/custom_widgets/pop_up_dialog_add_user.dart';
+import 'package:flutter_userapp_demo/custom_widgets/pop_up_dialog_filters.dart';
+import 'package:flutter_userapp_demo/constants/constants.dart';
+import 'package:flutter_userapp_demo/utility/utility.dart';
+
+void main() => runApp(MaterialApp(home:Application()));
 
-void main() {
-  runApp(MyApp());
+//Application
+class Application extends StatefulWidget {
+  @override
+  _ApplicationState createState() =>_ApplicationState();
 }
 
-class MyApp extends StatelessWidget {
-  // This widget is the root of your application.
+class _ApplicationState extends State<Application> {
+
+  final GlobalKey<_HomePageState> _mainScreenState = GlobalKey<_HomePageState>();
+  void applyFilters(HashMap<String,String> newFIlters)
+  {
+    _mainScreenState.currentState._applyFilters(newFIlters);
+  }
+
   @override
   Widget build(BuildContext context) {
     return MaterialApp(
-      title: 'Flutter Demo',
-      theme: ThemeData(
-        // This is the theme of your application.
-        //
-        // Try running your application with "flutter run". You'll see the
-        // application has a blue toolbar. Then, without quitting the app, try
-        // changing the primarySwatch below to Colors.green and then invoke
-        // "hot reload" (press "r" in the console where you ran "flutter run",
-        // or simply save your changes to "hot reload" in a Flutter IDE).
-        // Notice that the counter didn't reset back to zero; the application
-        // is not restarted.
-        primarySwatch: Colors.blue,
+      title: 'Welcome to Flutter',
+      home: Scaffold(
+        appBar: AppBar(
+          title: Text('User App'),
+          actions: [
+            new IconButton(
+                  icon: Icon(Icons.autorenew_sharp,size: 30,),
+                  onPressed: (){
+                     Navigator.of(context).push(MaterialPageRoute(builder: (context) => Application2()));
+                  }
+                ),
+            new Stack(
+              children:<Widget> [
+                new IconButton(
+                  icon: Icon(Icons.filter_alt_sharp,size: 30,),
+                  onPressed: (){
+                    showDialog(
+                      context: context,
+                      builder: (BuildContext context) => PopUpDialogFilters(
+                        context:context,
+                        callback:applyFilters,
+                        filters: _mainScreenState.currentState.filters
+                      ),
+                    );
+                  }
+                ),
+              ],
+            )
+            
+          ],
```

Comment:
This should be separated from here. Basically it's a screen widget. Therefore, we should name it with a Screen/Page suffix. 
Also, the packaging should be done to differentiate between screens and common widgets.

Reply:
Done sir, Please check once if it is right or not.